Add `FromScratchTrainingPlugin`

HamedHemati opened the pull request:
This PR adds a new plugin called FromScratchTraining that resets the model's weights and optimizer state after each training experience. It is used as a baseline for comparison with sequential fine-tuning in some papers. 

Minor: we can change the plugin's name if it's wordy. Another potential name could be: "StrategyResetter"

AntonioCarta commented on avalanche/training/plugins/from_scratch_training.py:
```diff
@@ -0,0 +1,54 @@
+from avalanche.models.dynamic_optimizers import reset_optimizer
+
+
+class FromScratchTrainingPlugin(SupervisedPlugin):
```
I think it's better to use the `SGDTemplate` here to support more use cases.

HamedHemati replied:
Agreed, will change it!

AntonioCarta commented:
Should we have a supervised strategy too? Since we have them for the other baseline methods.

HamedHemati commented:
> Should we have a supervised strategy too? Since we have them for the other baseline methods.

Updates:
- Updated the class definition
- Added strategy wrapper
- Added example

AntonioCarta commented:
Thanks!
